internal/lib/stats/cpu_metrics.go

```diff
@@ -0,0 +1,113 @@
+package statsserver
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cri-o/cri-o/internal/config/cgmgr"
+	"github.com/cri-o/cri-o/internal/lib/sandbox"
+	"github.com/cri-o/cri-o/internal/oci"
+	"github.com/sirupsen/logrus"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func GenerateSandboxCPUMetrics(sb *sandbox.Sandbox, c *oci.Container, stats interface{}) []*types.Metric {
+	cpu, ok := stats.(*cgmgr.CPUStats)
+	if !ok {
+		logrus.Errorf("Failed to assert stats as *cgmgr.CpuStats")
+		return nil
```

Comment:
If this ever fails, you will append `nil` into your metrics slice, and then it will be passed all over the place, before something tries to iterate over it and use the value to call some callable, and then it will explode, I believe.

Failing a type assertion might carry more weight than just a log line, I think.

Reply:
I added an if condition to guard against appending `nil` into metrics slice. I guess this should work?